diff --git a/panda_gym_jenga.egg-info/SOURCES.txt b/panda_gym_jenga.egg-info/SOURCES.txt
index 1c7b6e6..5b0ac47 100644
--- a/panda_gym_jenga.egg-info/SOURCES.txt
+++ b/panda_gym_jenga.egg-info/SOURCES.txt
@@ -12,6 +12,7 @@ panda_gym_jenga/envs/jenga_tasks.py
 panda_gym_jenga/envs/robot/__init__.py
 panda_gym_jenga/envs/robot/kinova.py
 panda_gym_jenga/envs/tasks/__init__.py
+panda_gym_jenga/envs/tasks/jenga_tower.py
 panda_gym_jenga/envs/tasks/jengapickandplace.py
 panda_gym_jenga/envs/tasks/simplejengapickandplace.py
 panda_gym_jenga/envs/tasks/stack3.py
diff --git a/scripts/test_ddpg_pickandplace.py b/scripts/test_ddpg_pickandplace.py
index a30af71..8d178bc 100644
--- a/scripts/test_ddpg_pickandplace.py
+++ b/scripts/test_ddpg_pickandplace.py
@@ -4,8 +4,8 @@ from stable_baselines3.her.goal_selection_strategy import GoalSelectionStrategy
 from stable_baselines3.common.monitor import Monitor
 from stable_baselines3.common.vec_env import DummyVecEnv, VecVideoRecorder
 from stable_baselines3.common.env_util import make_vec_env
-import gymnasium as gym
 import panda_gym_jenga
+import gymnasium as gym
 import wandb
 from wandb.integration.sb3 import WandbCallback
 config = {
diff --git a/wandb/latest-run b/wandb/latest-run
index 7bbdc38..92d541f 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20241220_153549-u7osx6vg
\ No newline at end of file
+run-20241223_111622-300c4ozk
\ No newline at end of file
